src/MatrixClient.cc

```diff
@@ -1021,20 +1027,19 @@ MatrixClient::makeUploadRequest(const QString &filename)
         endpoint.setPath(mediaApiUrl_ + "/upload");
         endpoint.setQuery(query);
 
-        QFile file(filename);
-        if (!file.open(QIODevice::ReadWrite)) {
-                qDebug() << "Error while reading" << filename;
+        if (!iodev->open(QIODevice::ReadOnly)) {
+                qDebug() << "Error while reading buffer" << iodev->errorString();
                 return nullptr;
         }
 
         QMimeDatabase db;
-        QMimeType mime = db.mimeTypeForFile(filename, QMimeDatabase::MatchContent);
+        QMimeType mime = db.mimeTypeForData(iodev.data());
 
         QNetworkRequest request(QString(endpoint.toEncoded()));
-        request.setHeader(QNetworkRequest::ContentLengthHeader, file.size());
+        request.setHeader(QNetworkRequest::ContentLengthHeader, iodev->size());
```

Comment:
Per [the `QIODevice::size` documentation](https://doc.qt.io/qt-5/qiodevice.html#size), this is incorrect if the device is not a random access device. The header should be omitted entirely if [`QIODevice::isSequential`](https://doc.qt.io/qt-5/qiodevice.html#isSequential) returns true.

[The spec](https://matrix.org/docs/spec/r0.0.0/client_server.html#post-matrix-media-r0-upload) doesn't seem to require the header at all, so I'm not sure why we're providing it in the first place. I believe it works fine without, so maybe it would be better to just remove this line entirely? Or is it of use to middleboxes sometimes?